File: java2csharp/java/Microsoft.Mobile.Engagement/engagement/reach/ContentStorage.java
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */

package com.microsoft.azure.engagement.reach;

import com.microsoft.azure.engagement.storage.EngagementStorage;

/** Constants used in SQLite */
final class ContentStorage
{
  /** Content local identifier */
  static final String OID = EngagementStorage.PRIMARY_KEY;

  /** Content reach campaign identifier */
  static final String CAMPAIGN_ID = "ci";

  /** DLC flag */
  static final String DLC = "dlc";

  /** Content DLC id */
  static final String DLC_ID = "id";

  /** Category */
  static final String CATEGORY = "cat";

  /** Delivery time */
  static final String DELIVERY_TIME = "dt";

  /** Campaign end time */
  static final String TTL = "ttl";

  /** Campaign end time user time zone flag */
  static final String USER_TIME_ZONE = "utz";

  /** Notification type */
  static final String NOTIFICATION_TYPE = "nt";

  /** Notification icon */
  static final String NOTIFICATION_ICON = "ic";

  /** Notification closeable */
  static final String NOTIFICATION_CLOSEABLE = "cl";

  /** Notification vibration */
  static final String NOTIFICATION_VIBRATION = "v";

  /** Notification sound */
  static final String NOTIFICATION_SOUND = "s";

  /** Notification title */
  static final String NOTIFICATION_TITLE = "tle";

  /** Notification message */
  static final String NOTIFICATION_MESSAGE = "msg";

  /** Notification big text */
  static final String NOTIFICATION_BIG_TEXT = "bt";

  /** Notification big picture */
  static final String NOTIFICATION_BIG_PICTURE = "bp";

  /** Action URL */
  static final String ACTION_URL = "au";

  /** Campaign JSON payload (dlc) */
  static final String PAYLOAD = "payload";

  /** Download identifier for attached file */
  static final String DOWNLOAD_ID = "download_id";

  /** Notification first displayed date */
  static final String NOTIFICATION_FIRST_DISPLAYED_DATE = "notification_first_displayed_date";

  /** Notification last displayed date */
  static final String NOTIFICATION_LAST_DISPLAYED_DATE = "notification_last_displayed_date";

  /** Notification actioned flag */
  static final String NOTIFICATION_ACTIONED = "notification_actioned";

  /** Content displayed flag */
  static final String CONTENT_DISPLAYED = "content_displayed";
}

File: java2csharp/csharp/Microsoft.Mobile.Engagement/engagement/reach/ContentStorage.cs
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */

namespace com.microsoft.azure.engagement.reach
{

	using EngagementStorage = com.microsoft.azure.engagement.storage.EngagementStorage;

	/// <summary>
	/// Constants used in SQLite </summary>
	internal sealed class ContentStorage
	{
	  /// <summary>
	  /// Content local identifier </summary>
	  internal const string OID = EngagementStorage.PRIMARY_KEY;

	  /// <summary>
	  /// Content reach campaign identifier </summary>
	  internal const string CAMPAIGN_ID = "ci";

	  /// <summary>
	  /// DLC flag </summary>
	  internal const string DLC = "dlc";

	  /// <summary>
	  /// Content DLC id </summary>
	  internal const string DLC_ID = "id";

	  /// <summary>
	  /// Category </summary>
	  internal const string CATEGORY = "cat";

	  /// <summary>
	  /// Delivery time </summary>
	  internal const string DELIVERY_TIME = "dt";

	  /// <summary>
	  /// Campaign end time </summary>
	  internal const string TTL = "ttl";

	  /// <summary>
	  /// Campaign end time user time zone flag </summary>
	  internal const string USER_TIME_ZONE = "utz";

	  /// <summary>
	  /// Notification type </summary>
	  internal const string NOTIFICATION_TYPE = "nt";

	  /// <summary>
	  /// Notification icon </summary>
	  internal const string NOTIFICATION_ICON = "ic";

	  /// <summary>
	  /// Notification closeable </summary>
	  internal const string NOTIFICATION_CLOSEABLE = "cl";

	  /// <summary>
	  /// Notification vibration </summary>
	  internal const string NOTIFICATION_VIBRATION = "v";

	  /// <summary>
	  /// Notification sound </summary>
	  internal const string NOTIFICATION_SOUND = "s";

	  /// <summary>
	  /// Notification title </summary>
	  internal const string NOTIFICATION_TITLE = "tle";

	  /// <summary>
	  /// Notification message </summary>
	  internal const string NOTIFICATION_MESSAGE = "msg";

	  /// <summary>
	  /// Notification big text </summary>
	  internal const string NOTIFICATION_BIG_TEXT = "bt";

	  /// <summary>
	  /// Notification big picture </summary>
	  internal const string NOTIFICATION_BIG_PICTURE = "bp";

	  /// <summary>
	  /// Action URL </summary>
	  internal const string ACTION_URL = "au";

	  /// <summary>
	  /// Campaign JSON payload (dlc) </summary>
	  internal const string PAYLOAD = "payload";

	  /// <summary>
	  /// Download identifier for attached file </summary>
	  internal const string DOWNLOAD_ID = "download_id";

	  /// <summary>
	  /// Notification first displayed date </summary>
	  internal const string NOTIFICATION_FIRST_DISPLAYED_DATE = "notification_first_displayed_date";

	  /// <summary>
	  /// Notification last displayed date </summary>
	  internal const string NOTIFICATION_LAST_DISPLAYED_DATE = "notification_last_displayed_date";

	  /// <summary>
	  /// Notification actioned flag </summary>
	  internal const string NOTIFICATION_ACTIONED = "notification_actioned";

	  /// <summary>
	  /// Content displayed flag </summary>
	  internal const string CONTENT_DISPLAYED = "content_displayed";
	}

}